File: benchmarks-QuTiP/expect_state.py
import qutip as qt
import numpy as np
import benchmarkutils

name = "expect_state"

samples = 5
evals = 100
cutoffs = range(5000, 150001, 5000)

def setup(N):
    op = (qt.destroy(N) + qt.create(N))
    psi = qt.Qobj(np.ones(N, complex)/(N**(1/2)))
    return op, psi

def f(op, psi):
    return qt.expect(op, psi)

print("Benchmarking:", name)
print("Cutoff: ", end="", flush=True)
checks = {}
results = []
for N in cutoffs:
    print(N, "", end="", flush=True)
    op, psi = setup(N)
    checks[N] = f(op, psi)
    t = benchmarkutils.run_benchmark(f, op, psi, samples=samples, evals=evals)
    results.append({"N": N, "t": t})
print()

benchmarkutils.check(name, checks)
benchmarkutils.save(name, results)

File: benchmarks-QuantumOptics.jl/expect_state.jl
using QuantumOptics
using BenchmarkTools
include("benchmarkutils.jl")


name = "expect_state"

samples = 5
evals = 1000
cutoffs = [5000:5000:100001;]

alpha = 0.7

function f(op, state)
    expect(op, state)
end

println("Benchmarking: ", name)
print("Cutoff: ")
results = []
for N in cutoffs
    print(N, " ")
    b = FockBasis(N-1)
    op = (destroy(b) + create(b))
    psi = Ket(b, ones(Complex128, N)/sqrt(N))
    t = @belapsed f($op, $psi) samples=samples evals=evals
    push!(results, Dict("N"=>N, "t"=>t))
end
println()

benchmarkutils.save(name, results)